client/dashboard/me/billing-purchases/change-payment-method.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+	purchaseQuery,
+	userSettingsQuery,
+	userPaymentMethodsQuery,
+	stripeConfigurationQuery,
+	razorpayConfigurationQuery,
+	queryClient,
+} from '@automattic/api-queries';
+import { RazorpayHookProvider } from '@automattic/calypso-razorpay';
+import { StripeHookProvider, useStripe } from '@automattic/calypso-stripe';
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { useNavigate } from '@tanstack/react-router';
+import { __experimentalVStack as VStack } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import { changePaymentMethodRoute } from '../../app/router/me';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { PaymentMethodSelector } from './payment-method-selector';
+import getPaymentMethodIdFromPayment from './payment-method-selector/get-payment-method-id-from-payment';
+import { useCreateAssignablePaymentMethods } from './payment-method-selector/use-create-assignable-payment-methods';
+import { getPurchaseUrl } from './urls';
+
+import './style.scss';
+
+function ChangePaymentMethod() {
+	const { purchaseId } = changePaymentMethodRoute.useParams();
+	const navigate = useNavigate();
+
+	const { data: purchase } = useSuspenseQuery( purchaseQuery( parseInt( purchaseId ) ) );
```

Comment:
Should this do an `isNaN(purchaseId)` check before passing it to `purchaseQuery`?